### PR TITLE
feat: immutable records app boilerplate

### PR DESCRIPTION
**Pull request by barthuijgen**
Immutable-records will be a new dApp meant for auctioning NFTs, as well as being an open source example of a dApp.

This MR just provides the boilerplate, meaning basically next-create-app setup. This is to get started with it and being able to already set up deployment and testing environments for further development and eventually golive.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (webpro) [inline .github/CODEOWNERS:27]: Please remove this or add someone (like @webpro), since you can't approve your own PR.

File: packages/apps/immutable-records/README.md
Comment (webpro) [inline packages/apps/immutable-records/README.md:34]: Please add this at the top like the other READMEs:

```
<!-- genericHeader start -->

<!-- genericHeader end -->

```

...and run `rush format`.

File: packages/apps/immutable-records/README.md
Comment (webpro) [inline packages/apps/immutable-records/README.md:28]: Not sure if we need this promotion here?

File: packages/apps/immutable-records/package.json
Comment (webpro) [inline packages/apps/immutable-records/package.json:19]: Types shouldn't be in `dependencies` (but `devDependencies` instead)

File: packages/apps/immutable-records/package.json
Comment (webpro) [inline packages/apps/immutable-records/package.json:22]: More devDeps here

File: packages/apps/immutable-records/public/next.svg
Comment (webpro) [inline packages/apps/immutable-records/public/next.svg:1]: Do we need these images?

File: packages/apps/immutable-records/src/app/page.module.css
Comment (webpro) [inline packages/apps/immutable-records/src/app/page.module.css:229]: Is this the _vanilla extract way?_ Or why is that dependency installed? :)

File: packages/apps/immutable-records/package.json
Comment (webpro) [inline packages/apps/immutable-records/package.json:11]: Please `format:md` like the others.

**Comment by webpro** [comment]
Please add a description, no idea what this is about?

File: .github/CODEOWNERS
Comment (barthuijgen) [reply .github/CODEOWNERS:27]: Approving your own PR is not the purpose of CODEOWNERS, it's to approve when someone else makes changes to that code right? But I'll remove it.

File: packages/apps/immutable-records/package.json
Comment (barthuijgen) [reply packages/apps/immutable-records/package.json:11]: This gives me the following, I'll leave it out for now bug can investigate later.
image

Comment (webpro) [reply packages/apps/immutable-records/package.json:12]: You can add `"@kadena-dev/markdown": "workspace:*",` to devDeps. It's more correct to also add `remark-cli`, but you can omit it for now.

File: packages/apps/immutable-records/build.mjs
Comment (webpro) [inline packages/apps/immutable-records/build.mjs:36]: Warnings are annoying, but not worth a workaround like this. Do we need this to let the build succeed in CI? I have a hunch this is caused by https://github.com/kadena-community/kadena.js/blob/main/common/config/rush/command-line.json#L165-L185

**Review by webpro — APPROVED** [review]

File: packages/apps/immutable-records/README.md
Comment (webpro) [reply packages/apps/immutable-records/README.md]: This is not resolved? In general I think comments/issues in PRs should be marked as resolved by the reviewer, not the pull-requester.

Comment (barthuijgen) [reply packages/apps/immutable-records/README.md]: Right, sorry about that. I assumed it would be pretty straightforward since I added the genericHeader as other apps have it. How is it not resolved yet?

File: packages/apps/immutable-records/build.mjs
Comment (barthuijgen) [reply packages/apps/immutable-records/build.mjs:36]: Ah, that does look like exactly the properly we would need to solve this! indeed much better then a custom startup script.

I am wondering if there is a way to enable this for only the "@kadena/immutable-records" app? From what I see this is configured once for the whole repo. That would not be my intention.

Would you rather have this setting changed for the `build` command in the entire repo, or keep the single-app script?

Comment (barthuijgen) [reply packages/apps/immutable-records/build.mjs:36]: Looking at it more thoroughly, there doesn't seem to be a build command in `commnad-line.json`, so I'm not sure how `allowWarningsInSuccessfulBuild` would be added.

File: packages/apps/immutable-records/README.md
Comment (barthuijgen) [reply packages/apps/immutable-records/README.md]: Thanks, I did notice format had some additional changes. It actually deletes the description under the `# @kadena/immutable-records` header, not sure if that is intentional. I moved it to below the generic header now.

Comment (webpro) [reply packages/apps/immutable-records/README.md]: It uses basically this template:

```
# {pkg.name}

{pkg.description}

<img>
```

Definitely a part that's lacking in our docs. I'll see what I can do about that. Thanks for bearing with me!

**Pull request merged**
